arduino-ide-extension/src/browser/contributions/cloud-contribution.ts

```diff
@@ -0,0 +1,121 @@
+import { CompositeTreeNode } from '@theia/core/lib/browser/tree';
+import { nls } from '@theia/core/lib/common/nls';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { CreateApi } from '../create/create-api';
+import { CreateFeatures } from '../create/create-features';
+import { CreateUri } from '../create/create-uri';
+import { Create, isNotFound } from '../create/typings';
+import { CloudSketchbookTree } from '../widgets/cloud-sketchbook/cloud-sketchbook-tree';
+import { CloudSketchbookTreeModel } from '../widgets/cloud-sketchbook/cloud-sketchbook-tree-model';
+import { CloudSketchbookTreeWidget } from '../widgets/cloud-sketchbook/cloud-sketchbook-tree-widget';
+import { SketchbookWidget } from '../widgets/sketchbook/sketchbook-widget';
+import { SketchbookWidgetContribution } from '../widgets/sketchbook/sketchbook-widget-contribution';
+import { SketchContribution } from './contribution';
+
+export function sketchAlreadyExists(input: string): string {
+  return nls.localize(
+    'arduino/cloudSketch/alreadyExists',
+    "Cloud sketch '{0}' already exists.",
+    input
+  );
+}
+export function sketchNotFound(input: string): string {
+  return nls.localize(
+    'arduino/cloudSketch/notFound',
+    "Could not pull the cloud sketch '{0}'. It does not exist.",
```

Comment:
Should "cloud" be capitalized?

I'm not a fan of the recent trend I've observed of sometimes capitalizing "sketch" in sentences, since that is not a proper noun, but if "cloud" is short for [the "**Arduino Cloud**" service](https://cloud.arduino.cc/) (as opposed to being a reference to the general concept of "the cloud"), then this is a proper noun and so should be capitalized.

Reply:
> Should "cloud" be capitalized?

I will leave this decision to you. Whichever you think is the best.

> the recent trend I've observed of sometimes capitalizing "sketch" in sentences

I found the following translations from this PR where the "sketch" is capitalized in a sentence. (I hope I did not leave out anything):

*These have not changed in this PR or recently:*

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/widgets/cloud-sketchbook/cloud-sketchbook-tree.ts#L107

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/dialogs/cloud-share-sketch-dialog.tsx#L66

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/dialogs/cloud-share-sketch-dialog.tsx#L75

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/dialogs/cloud-share-sketch-dialog.tsx#L86

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/widgets/cloud-sketchbook/cloud-sketchbook-tree.ts#L70

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/dialogs/firmware-uploader/firmware-uploader-component.tsx#L187

----

*These have not changed, and the sentence starts with "sketch":*

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/contributions/debug.ts#L215

----

*These are command or menu labels, so capitalized "sketch" is expected:*

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/contributions/new-sketch.ts#L24

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/widgets/cloud-sketchbook/cloud-sketchbook-composite-widget.tsx#L59

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/contributions/open-sketch-external.ts#L25

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/widgets/sketchbook/sketchbook-commands.ts#L15

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/arduino-frontend-contribution.tsx#L161

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/contributions/open-sketch.ts#L93

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/contributions/archive-sketch.ts#L25

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/widgets/cloud-sketchbook/cloud-sketchbook-contributions.ts#L78

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/widgets/cloud-sketchbook/cloud-sketchbook-contributions.ts#L87

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/contributions/new-cloud-sketch.ts#L55

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/widgets/sketchbook/sketchbook-composite-widget.tsx#L84

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/widgets/cloud-sketchbook/cloud-sketchbook-contributions.ts#L217

----

*These are new and were requested in the design doc:*

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/contributions/rename-cloud-sketch.ts#L85

https://github.com/arduino/arduino-ide/blob/9a3b1079aa5e375f563df3055cd1885a2f84c440/arduino-ide-extension/src/browser/contributions/new-cloud-sketch.ts#L100

Let me know what to do. Thanks!

Reply:
>I will leave this decision to you. Whichever you think is the best.

I surveyed the Arduino Cloud-related content on arduino.cc and found that "Cloud" is capitalized in contexts equivalent to "Cloud sketch". So I think it should be capitalized.

>the sentence starts with "sketch":

>These are command or menu labels, so capitalized "sketch" is expected:

Yes, in cases where the work occurs at the start of a sentence or contexts where the standard for capitalization style is "[title case](https://en.wikipedia.org/wiki/Title_case)", there is no question that "Sketch" is correct.

>These are new and were requested in the design doc:

Then I guess I would have to request the input of the designers.

@91volt @gmarchiarduino please provide clear rules for when "sketch" should and should not be capitalized. As I've said several times in the past, it would be very helpful to have this sort of thing documented in a style guide.

Reply:
This part is something I cannot control, so I moved it to its dedicated task: https://github.com/arduino/arduino-ide/issues/1884.